python/ray/rllib/a3c/base_evaluator.py

```diff
@@ -73,20 +77,51 @@ def get_weights(self):
     def set_weights(self, params):
         self.policy.set_weights(params)
 
-    def update_filters(self, obs_filter=None, rew_filter=None):
-        if rew_filter:
-            # No special handling required since outside of threaded code
-            self.rew_filter = rew_filter.copy()
-        if obs_filter:
-            self.sampler.update_obs_filter(obs_filter)
-
     def save(self):
+        filters = self.get_filters(flush_after=True)
         weights = self.get_weights()
-        return pickle.dumps({"weights": weights})
+        return pickle.dumps({
+            "filters": filters,
+            "weights": weights})
 
     def restore(self, objs):
         objs = pickle.loads(objs)
+        self.sync_filters(**objs["filters"])
         self.set_weights(objs["weights"])
 
+    def sync_filters(self, obs_filter=None, rew_filter=None):
```

Comment:
Should this stuff be part of filter manager?

Reply:
I figured the filter manager should sit on the driver and should manage coordination among workers. One could extend it to also sit on each evaluator and manage the synchronization there too, but you would still need to expose getters and setters on the evaluators. I'm not sure what the right design is.